CONTRIBUTING.md

```diff
@@ -179,6 +179,22 @@ $ git commit
 $ git push fork feature
 ```
 
+### Commit Messages
+
+Use descriptive commit messages. Here are [some recommendations](https://cbea.ms/git-commit/)
+on how to write good commit messages.
+When creating PRs GitHub will automatically copy commit messages into the PR description,
+so it is a useful habit to write good commit messages before the PR is created.
+Also, unless you actually want to tell a story with multiple commits make sure to squash
+into a single commit before creating the PR.
+
+When merging PRs with multiple commits, they will be squashed and GitHub will
+concatenate all commit messages right before you hit the "Confirm squash and merge"
+button. Make sure to edit this concatenated message to make it right before merging.
```

Comment:
Only maintainers can edit this message, so maybe this should not be on CONTRIBUTING.md?

Reply:
Do we have a better place for this? I don't think we have a "maintainer's only" secret contributing guide. :-)

Reply:
> I don't think we have a "maintainer's only" secret contributing guide. :-)

We don't and we shouldn't :) I don't have a good idea of where it should go, it just feels weird to have something that applies to just a handful of people on the doc that everybody reads to get started contributing

Reply:
I think we should just clarify that it is a message for maintainers (e.g. Maintainers, when merging PRs, should squash commits and edit the commit message to concisely describe the PR)

Reply:
>I think we should just clarify that it is a message for maintainers (e.g. Maintainers, when merging PRs, should squash commits and edit the commit message to concisely describe the PR)
Done.